Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -323,147 +413,269 @@ function Enable-StorageEncryptionInTransit
+                    $logResource.Add("Reason","Unsuccessful in enabling secure transfer on the Storage Account.")    
+                    $logSkippedResources += $logResource
+                    Write-Host "Unsuccessful in enabling secure transfer on the Storage Account." -ForegroundColor $([Constants]::MessageType.Warning) 
+                    Write-Host $([Constants]::SingleDashLine)
                 }
+            }        
+            catch
+            {
+                Write-Host "Error occurred while enabling secure transfer on the Storage Account." -ForegroundColor $([Constants]::MessageType.Error) 
+                Write-Host "Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::SingleDashLine)
+                $remediationFailure += $_
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.StorageAccountName))
+                $logResource.Add("Reason","Error occurred while enabling secure transfer on the Storage Account.")    
+                $logSkippedResources += $logResource
             }
-            else
+        }
+        Write-Host $([Constants]::DoubleDashLine)
+        if($AutoRemediation)
+        {
+            if($($remediationSuccess|Measure-Object).Count -gt 0)
             {
-                Write-Host " $([Constants]::SingleDashLine)"
-
-                # Creating the log file
-                Write-Host "Exporting configurations of Storage Account(s) that don't have secure transfer enabled. You may want to use this CSV as a pre-check before actual remediation." 
-                $stgWithDisableHTTPS | Export-CSV -Path "$($folderpath)\StorageWithDisableHTTPS.csv" -NoTypeInformation
-                Write-Host "Path: $($folderPath)\StorageWithDisableHTTPS.csv" -ForegroundColor $([Constants]::MessageType.Info)
-            }          
-        }     
-        else
+                # Write this to a file.
+                $storageAccountRemediatedFile = "$($backupFolderPath)\RemediateStorageAccountsEnableSecureTransfer.csv"
+                $remediationSuccess | Export-CSV -Path $storageAccountRemediatedFile -NoTypeInformation
+                Write-Host "This information has been saved to [$($storageAccountRemediatedFile)]." -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host "`nUse this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+            
+            if($($remediationFailure|Measure-Object).Count -gt 0)
+            {
+                # Write this to a file.
+                $storageAccountSkippedFile = "$($backupFolderPath)\SkippedStorageAccountsEnableSecureTransfer.csv"
+                $remediationFailure | Export-CSV -Path $storageAccountSkippedFile -NoTypeInformation
```

Comment:
One line summary like this is the list of errored resource will help to set context for the end user.

Reply:
Same as above